test/AddressRegistry.Tests/BackOffice/Lambda/Infrastructure/MessageHandlerTests.cs

```diff
@@ -331,47 +331,47 @@ public async Task WhenSqsAddressDeregulateRequest_ThenSqsLambdaAddressDeregulate
         }
 
         [Fact]
-        public async Task WhenSqsAddressProposeRequest_ThenSqsLambdaAddressProposeRequestIsSent()
+        public async Task WhenProposeRequest_ThenProposeRequestIsSent()
         {
             // Arrange
             var mediator = new Mock<IMediator>();
             var containerBuilder = new ContainerBuilder();
             containerBuilder.Register(_ => mediator.Object);
             var container = containerBuilder.Build();
 
-            var messageData = Fixture.Create<SqsAddressProposeRequest>();
+            var expectedRequest = Fixture.Create<ProposeSqsRequest>();
             var messageMetadata = new MessageMetadata { MessageGroupId = Fixture.Create<string>() };
 
             var sut = new MessageHandler(container);
 
             // Act
             await sut.HandleMessage(
-                messageData,
+                expectedRequest,
                 messageMetadata,
                 CancellationToken.None);
 
             // Assert
             mediator
-                .Verify(x => x.Send(It.Is<SqsLambdaAddressProposeRequest>(request =>
-                    request.TicketId == messageData.TicketId
-                    && request.MessageGroupId == messageMetadata.MessageGroupId
-                    && request.Request == messageData.Request
-                    && request.IfMatchHeaderValue == null
-                    && request.Provenance == messageData.ProvenanceData.ToProvenance()
-                    && request.Metadata == messageData.Metadata
+                .Verify(x => x.Send(It.Is<ProposeLambdaRequest>(actualRequest =>
+                    actualRequest.TicketId == expectedRequest.TicketId
+                    && actualRequest.MessageGroupId == messageMetadata.MessageGroupId
+                    && actualRequest.Request == expectedRequest.Request
+                    && actualRequest.IfMatchHeaderValue == expectedRequest.IfMatchHeaderValue
```

Comment:
Note that I had to change the assertion here because Fixture.Create<ProposeSqsRequest> was filling in the ifmatchheader, I don't understand why it was **null** before.

Reply:
It is null because you are creating an address. So no previous etag/event hash exists.